### DISCUSSION TITLE
Is submariner compatible with kubeadm?

### DISCUSSION THREAD

**Discussion by pncps99** [submariner-io/submariner#1890] [Q&A]
I am planning on doing an installation of submariner with kubeadm, and I was considering if there are any compatibility issues. If such an installation is possible, what is the preferred CNI plugin? I like to use Flannel or Calico.

**Comment by sridhargaddam** [comment]
Submariner should generally work with K8s clusters that use kube-proxy with iptables mode. In terms of CNIs, the list of supported CNIs are listed here - https://submariner.io/getting-started/#support-matrix

Both Flannel and Calico are in the validated list. So, you can use either of them.
